File: FinanceApp.Web/src/models/TransactionDtos/UpdateTransactionDto.ts
import { TransactionTypeEnum } from "../Enums/TransactionType.enum";
import { Money } from "../Money/Money";

export interface UpdateTransactionDto {
    id: string,
    name: string,
    description?: string,
    value: Money,
    transactionType: TransactionTypeEnum,
    transactionDate?: Date,
    transactionGroupId: string
}

File: FinanceApp.Application/Dtos/TransactionDtos/UpdateTransactionDto.cs
using FinanceApp.Domain.Entities;

namespace FinanceApp.Application.Dtos.TransactionDtos;

public class UpdateTransactionDto
{
  public Guid Id { get; set; }
  public string Name { get; set; } = string.Empty;
  public string? Description { get; set; }
  public Money Value { get; set; } = new();
  public TransactionTypeEnum TransactionType { get; set; }
  public DateTimeOffset TransactionDate { get; set; }
  public Guid? TransactionGroupId { get; set; } = null!;
}
